src/view.ts

```diff
@@ -319,6 +323,22 @@ export class WriteableStickyNote extends AbstractStickyNote {
   set note(note: Note | undefined) {
     this.updating = true;
     super.note = note;
+    if (note?.uuid !== this.note?.uuid) {
+      this.add_modified_listener();
+    }
+  }
+
+  add_modified_listener() {
+    if (!this.note) return;
+    if (this.source) GLib.source_remove(this.source);
+    this.source = this.buffer.connect("modified-changed", (buffer) => {
+      if (buffer.get_modified()) {
+        save_note(this.note!)
+          .then(() => {
+            this.buffer.set_modified(false);
+          });
+      }
+    });
```

Comment:
:+1:

Reply:
also should be noted that we can't save during "vfunc_shutdown" as it doesn't wait for promises ("save" is a promise now) to resolve. but that doesn't matter anyways since the files are saved on each click